primitives/keystore/src/lib.rs

```diff
@@ -194,6 +194,20 @@ pub trait CryptoStore: Send + Sync {
 		public: &sr25519::Public,
 		transcript_data: VRFTranscriptData,
 	) -> Result<Option<VRFSignature>, Error>;
+
+	/// Sign pre-hashed
+	///
+	/// Signs a pre-hashed message with the private key that matches
+	/// the ECDSA public key passed.
+	///
+	/// Returns the SCALE encoded signature if key is found and supported,
+	/// `None` if the key doesn't exist or an error when something failed.
+	async fn ecdsa_sign_prehashed(
+		&self,
+		id: KeyTypeId,
+		public: &ecdsa::Public,
+		msg: &[u8; 32],
+	) -> Result<Option<Vec<u8>>, Error>;
```

Comment:
Why is the signature SCALE-encoded? With `VRFSignature` we have concrete type and I think we should return `ecdsa::Signature` here.

Reply:
Yeah. We should return the concrete type here.